src/test/java/uk/gov/pay/adminusers/resources/ServiceResourceCreateTest.java

```diff
@@ -43,26 +39,19 @@ public void shouldSuccess_whenCreateAServiceWithValidGatewayAccounts() throws Ex
                 .put("gateway_account_ids", new String[]{"1", "2"})
                 .build();
 
-        ValidatableResponse validatableResponse = givenSetup()
+        givenSetup()
                 .when()
                 .accept(JSON)
                 .body(mapper.writeValueAsString(payload))
                 .post("/v1/api/services")
                 .then()
-                .statusCode(201);
-
-        validatableResponse
+                .statusCode(201)
                 .body("name", is("System Generated"))
-                .body("external_id", notNullValue());
-
-        String serviceExternalId = validatableResponse.extract().jsonPath().getString("external_id");
-
-        List<Map<String, Object>> gatewayAccountsForService = databaseHelper.findGatewayAccountsByService(serviceExternalId);
-        List<String> gatewayAccountIds = gatewayAccountsForService.stream()
-                .map(gaEntry -> gaEntry.get("gateway_account_id").toString()).collect(toList());
+                .body("external_id", notNullValue())
+                .body("gateway_account_ids", hasSize(2))
+                .body("gateway_account_ids[0]", is("1"))
```

Comment:
Is `gateway_account_ids` elements order is dictated by request? Probably it is better to have the sorting mechanism implemented in https://github.com/alphagov/pay-adminusers/blob/master/src/main/java/uk/gov/pay/adminusers/persistence/entity/ServiceEntity.java#L83, for example https://github.com/alphagov/pay-adminusers/blob/master/src/main/java/uk/gov/pay/adminusers/persistence/entity/UserEntity.java#L218